internal/framework/status/updater.go

```diff
@@ -169,3 +182,43 @@ func (upd *updaterImpl) update(
 			"kind", obj.GetObjectKind().GroupVersionKind().Kind)
 	}
 }
+
+func UpdateObjStatus(
```

Comment:
does this function need to be exported?

Reply:
Removed because we use imported retrying function.